### PR TITLE
Allow for tracking session to be resumed (inc. across QField restarts)

### PR DESCRIPTION
**Pull request by nirvn**
This PR adds the infrastructure to resume tracking sessions (including across QField reboots/restarts). When a tracking session has previously been started, QField now offers to resume the last session:

image

Practically, it means that for lines and polygons, the created feature from the previous session is carried over.

In order to be able to restore tracking sessions _across QField launches_, or pause/resume tracks within a same session, we need to decouple the tracking settings UI from the tracking session item. Doing so actually makes the handling of tracks a little bit easier to understand, and avoids mixing UI and core elements together.

The newly added TrackerSettings.qml is essentially the popup part of TrackingSession.qml taken out.

I've taken the time to clean up the tracking model a fairbit and add missing data() roles as well as guarding against invalid row index.

### CODE REVIEW

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for c3a80926c57a17e327435495e249dc726ff0ccef: [**arm64-android**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-4590-fbcd74608f774d7652f9fe3604b87542a8c1cf13-arm64-android.apk)
<details>
<summary>Other architectures</summary>

- [arm-neon-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-4590-fbcd74608f774d7652f9fe3604b87542a8c1cf13-arm-neon-android.apk)
- [x64-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-4590-fbcd74608f774d7652f9fe3604b87542a8c1cf13-x64-android.apk)
- [x86-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-4590-fbcd74608f774d7652f9fe3604b87542a8c1cf13-x86-android.apk)
- [**all access arm64-android**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield_all_access-dev-4590-fbcd74608f774d7652f9fe3604b87542a8c1cf13-arm64-android.apk)
- [all access arm-neon-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield_all_access-dev-4590-fbcd74608f774d7652f9fe3604b87542a8c1cf13-arm-neon-android.apk)
- [all access x64-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield_all_access-dev-4590-fbcd74608f774d7652f9fe3604b87542a8c1cf13-x64-android.apk)
- [all access x86-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield_all_access-dev-4590-fbcd74608f774d7652f9fe3604b87542a8c1cf13-x86-android.apk)
</details>

File: src/core/geometry.cpp
Comment (suricactus) [inline src/core/geometry.cpp:99]:
```suggestion
    if ( mVectorLayer )
    {
      mRubberbandModel->setDataFromGeometry( geometry, mVectorLayer->crs() );
    }
    else
    {
      mRubberbandModel->setDataFromGeometry( geometry );
    }
```

File: src/core/projectinfo.h
Comment (suricactus) [inline src/core/projectinfo.h:68]: Docs

File: src/core/projectinfo.h
Comment (suricactus) [inline src/core/projectinfo.h:107]: Docs

File: src/core/tracker.h
Comment (suricactus) [inline src/core/tracker.h:100]: Docs

File: src/qml/TrackerSettings.qml
Comment (suricactus) [inline src/qml/TrackerSettings.qml:330]:
```suggestion
            // Setting the distance to `NaN` is ok if text is not parsable
            positioningSettings.trackerErroneousDistance = parseFloat( text )
```

File: src/qml/TrackerSettings.qml
Comment (suricactus) [inline src/qml/TrackerSettings.qml:263]:
```suggestion
                ? qsTr( "When the sensor constraint is activated alone, vertex additions will occur whenever the sensor has captured new data." )
```
When reading I feel like there is a missing article "the", but my English is not perfect.

File: src/qml/TrackerSettings.qml
Comment (suricactus) [inline src/qml/TrackerSettings.qml:250]:
```suggestion
          text: qsTr( "When enabled, vertices will only be recorded when all active constraints are met. If the setting is disabled, individual constraints met will trigger a vertex addition." )
```

I think the first sentence is fixed, but the second one still does not make sense to me.

File: src/qml/TrackerSettings.qml
Comment (suricactus) [inline src/qml/TrackerSettings.qml:202]: What exactly are "sensor constraints"? Down there we discuss just "constraints" are they the same?

File: src/qml/TrackerSettings.qml
Comment (suricactus) [inline src/qml/TrackerSettings.qml:139]: Thi sis super repetitive, no idea if we can abstract it in any means?

File: src/qml/TrackerSettings.qml
Comment (suricactus) [inline src/qml/TrackerSettings.qml:138]:
```suggestion
          onTextChanged: {
            // Storing `NaN` values is ok
            positioningSettings.trackerTimeInterval = parseFloat( text )
          }
```

File: src/qml/TrackerSettings.qml
Comment (suricactus) [inline src/qml/TrackerSettings.qml:198]: Again

**Comment by suricactus** [comment]
Few comments on the UI:
- I would make the popup without margins on the side or fullscreen. At least on vertical mobile is no nice.
- The groupings Time/Distance/Sensor constraints are not supe clear. Why don't we use tabs or another way of grouping? The distance constraints settings that appear after enabling the toggle are not indented. Probably adding bold on the parent groups will improve the situation?
- Without documentation for me it is super unclear what the terms are meaning. E.g. in the QField Settings things do have meaning. But I am not sure if the Tracking Settings use coherent terms as the QField Settings. 

@lindacamathias can you test before me merge all of this, I am pretty sure you are more familiar with the context and will provide better user feedback.

**Review by suricactus — COMMENTED** [review]
Some styling comments

**Comment by suricactus** [comment]
Few comments on the UI:
- I would make the popup without margins on the side or fullscreen. At least on vertical mobile is no nice.
- The groupings Time/Distance/Sensor constraints are not supe clear. Why don't we use tabs or another way of grouping? The distance constraints settings that appear after enabling the toggle are not indented. Probably adding bold on the parent groups will improve the situation?
- Without documentation for me it is super unclear what the terms are meaning. E.g. in the QField Settings things do have meaning. But I am not sure if the Tracking Settings use coherent terms as the QField Settings. 

@lindacamathias can you test before me merge all of this, I am pretty sure you are more familiar with the context and will provide better user feedback.

File: src/qml/TrackerSettings.qml
Comment (nirvn) [reply src/qml/TrackerSettings.qml:250]: Tried to improve by adding a word :) what it means is that say you have a time constraint of 5 sec and a distance constraint of 10 meters, a vertex will be added when individual constraints "turn green", i.e. when the time goes above 5 sec of when the distance goes above 10 meters.

File: src/qml/TrackerSettings.qml
Comment (nirvn) [reply src/qml/TrackerSettings.qml:202]: @suricactus , I've updated to "activate new sensor data constraint".

**Comment by nirvn** [comment]
@suricactus , thanks for the review, I've added a bunch of missing documentation in projectinfo.h on top of the functions hinted at in your review.

Regarding popup styling (and other possible UI improvements), I suggest we defer to another PR, this one is quite disruptive as it is. I don´t think we'd go fullcreen on this one as for tablets it'd look weird for both orientation, however we can reduce margins for narrow screen width.

I'd be -1 on tabs in this UI in part because I think we want to keep most settings super quick to access. I have been trying to think of a way to visually regroup settings into themes (e.g. her we could have Constraints and General groups), it'd be useful here and in the settings panel. I haven't settled on the way it should be done visually yet though.

I'm seeing margins for constraints and erroneous position sub-parameters here.

If you have a photo of the bottom padding issue on Qt6, could you attach here?

**Comment by nirvn** [comment]
(note that the client sponsoring this feature has done some testing already, which revealed a couple of regressions that have been addressed)

**Comment by suricactus** [comment]
Thanks a lot for the PR and speedy update of the UI. Looks good now!

**Review by suricactus — APPROVED** [review]
Thanks a lot for the PR and speedy update of the UI. Looks good now!

**Comment by nirvn** [comment]
@suricactus , thanks for the crucial pair of eyes and suggestions, made a huge difference.

**Pull request merged**
